File: generator.js
function ascii_check(value) {
    if (value < 32) {
      return value + 32;
    }
  
    if (value === 127) {
      return value - 1;
    }
  
    return value;
  }
  
  function cycle(keyword) {
    var head;
    [head, ...tail] = keyword;
    tail += head;
    return tail;
  }
  
  function list_to_string(char_list) {
    var new_string;
    new_string = "";
  
    for (var element, _pj_c = 0, _pj_a = char_list, _pj_b = _pj_a.length; _pj_c < _pj_b; _pj_c += 1) {
      element = _pj_a[_pj_c];
      new_string.join(element);
    }
  
    return new_string;
  }
  
  function encode(letter, keyword) {
    var asc_num;
    asc_num = ord(letter) * ord(keyword[0]) % 128;
    asc_num = ascii_check(asc_num);
    keyword = cycle(keyword);
    return [chr(asc_num), keyword];
  }
  
  function encrypt(plaintext, ciphertext, keyword) {
    var chr, letter;
  
    if (plaintext.length === 0) {
      return ciphertext;
    }
  
    [letter, ...tail] = plaintext;
    [chr, keyword] = encode(letter, keyword);
    ciphertext += chr;
    return encrypt(tail, ciphertext, keyword);
  }
  
  function extend(password) {
    var extension
    if (password.length < 16) {
        var splitString = password.split("");
        var reverseArray = splitString.reverse();
        extension = reverseArray.join("");

        
      return password + extension;
    }
  }

  console.log(encrypt(extend("password"), "", "google.com"));

File: generator.py
# pylint: disable=missing-module-docstring
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring



# Checks and adjusts for valid ASCII characters

def ascii_check(value):
    if value < 32:
        return value + 32
    if value == 127:
        return value - 1
    return value

# Cycles through string and places current head at the tail.

def cycle(keyword):
    head, *tail = keyword
    tail += head
    return tail

# Converts single character of plaintext string into cipher text character

def encode(letter, keyword):
    asc_num = (ord(letter) * ord(keyword[0])) % 128
    asc_num = ascii_check(asc_num)
    keyword = cycle(keyword)
    return chr(asc_num), keyword

# Recursively loops through plaintext string to convert and return full ciphertext string

def encrypt(plaintext, ciphertext, keyword):
    if len(plaintext) == 0:
        return ciphertext

    letter, *tail = plaintext
    char, keyword = encode(letter, keyword)
    ciphertext += char
    return encrypt(tail, ciphertext, keyword)

# Returns mirrored plaintext extension if length condition is not met.

def extend(password):
    if len(password) < 16:
        password += password[::-1]
    return password

print(encrypt(extend("password"),"","google.com"))